### DISCUSSION TITLE
Question about high relative F RMSE

### DISCUSSION THREAD

**Discussion by treasureUu** [ACEsuit/mace#899] [Q&A] [answered]
Hi, 
Thank you everyone for looking at my question and helping me. 
I am currently training a fine-tuning model, but **the relative F RMSE % in the Error-table is very high**, like this: 

```
INFO: Error-table on TRAIN and VALID:
+---------------+---------------------+------------------+-------------------+
|  config_type  | RMSE E / meV / atom | RMSE F / meV / A | relative F RMSE % |
+---------------+---------------------+------------------+-------------------+
| train_default |            2.9      |          8.0     |        318.11     |
| valid_default |            3.2      |          9.0     |        360.86     |
+---------------+---------------------+------------------+-------------------+
INFO: Evaluating Default_default ...
INFO: Error-table on TEST:
+-----------------+---------------------+------------------+-------------------+
|   config_type   | RMSE E / meV / atom | RMSE F / meV / A | relative F RMSE % |
+-----------------+---------------------+------------------+-------------------+
| Default_default |            3.6      |          8.9     |        351.39     |
+-----------------+---------------------+------------------+-------------------+
```

To better describe my dataset, I have copied the information about the **dataset** from the log file. 

```
INFO: Using isolated atom energies from training file
INFO: Training set [416 configs, 416 energy, 69888 forces] loaded from 'Cr_dof_optim_training_less.xyz'
INFO: Validation set [64 configs, 64 energy, 10752 forces] loaded from 'optim_valid_shuffled_301-350andmore.xyz'
INFO: Test set (57 configs) loaded from 'optim_test_shuffled_351-400andmore.xyz':
INFO: Default_default: 57 configs, 57 energy, 9576 forces
```

I have also attached the parameters used for training (**yml file**). 
```
name: crystal_Cr_dof_optim_isolated_atom_FEwight_Finetuning
train_file: Cr_dof_optim_training_less.xyz
device: cuda
enable_cueq: True
valid_file: optim_valid_shuffled_301-350andmore.xyz
test_file: optim_test_shuffled_351-400andmore.xyz
batch_size: 32
valid_batch_size: 32
multiheads_finetuning: False
max_num_epochs: 4046
lr: 0.0005
seed: 31
ema: True
ema_decay: 0.99
forces_weight: 10
energy_weight: 100
energy_key: energy_hientropy
forces_key: forces_hientropy
stress_key: stress_hientropy
foundation_model: /scratch/user/djy/mace-models/2023-12-03-mace-128-L1_epoch-199.model
```

I am unsure which parameters need adjustment or where the potential issues lie. 
Could someone please help me in identifying any possible issues? I would greatly appreciate your help. Thank you very much.

**Comment by gabor1** [comment, marked as answer]
Your absolute force errors are rather small (less than 10 meV/Å) so your high relative force errors mean that your forces are very small. Probably because you have geometry optimised configurations. So in this case there is nothing wrong with high relative force errors. But we recommend not to train on optimised geometries. You should train on geometries that have decent sized forces, typical of those you will encounter when you use the potential. Keep your optimised structures as an independent test set.

**Reply by treasureUu** [reply]
> Your absolute force errors are rather small (less than 10 meV/Å) so your high relative force errors mean that your forces are very small. Probably because you have geometry optimised configurations. So in this case there is nothing wrong with high relative force errors. But we recommend not to train on optimised geometries. You should train on geometries that have decent sized forces, typical of those you will encounter when you use the potential. Keep your optimised structures as an independent test set.

Thank you very much for your help; your assessment was absolutely correct. The dataset I've been using indeed consists of optimized structures with very small forces. I will follow your advice and prepare a new dataset with appropriately applied forces. ♥

**Reply by treasureUu** [reply]
> Your absolute force errors are rather small (less than 10 meV/Å) so your high relative force errors mean that your forces are very small. Probably because you have geometry optimised configurations. So in this case there is nothing wrong with high relative force errors. But we recommend not to train on optimised geometries. You should train on geometries that have decent sized forces, typical of those you will encounter when you use the potential. Keep your optimised structures as an independent test set.

Thank you for your kind guidance.

I've encountered another challenge.
My dataset consists of **two-dimensional materials**, with DFT calculations performed using VASP and a vacuum layer of 15 Ångstrom. I applied a force with a **strain of 10%**, but the **forces on each atom** in the OUTCAR are still very small, for example, 0.002795 (eV/Ångst). 

In this situation, if I use stress on the cell and increase the **stress_weight** parameter, would this approach help in convergence?

I would appreciate your help on this matter.

**Comment by ilyes319** [comment]
Can you please share your full log file so I can check? you might be using incorrect keys.

**Reply by treasureUu** [reply]
> Can you please share your full log file so I can check? you might be using incorrect keys.

Yes, the log file is attached. That's helpful. 

Also, I want to introduce my dataset to better identify any issues. It consists of **two-dimensional materials**, with DFT calculations performed using VASP and a vacuum layer of 15 Ångstrom. 

If you have any suggestions, I would greatly appreciate it. Thank you.



[macedjy.txt](https://github.com/user-attachments/files/19598885/macedjy.txt)

**Reply by ilyes319** [reply]
Can you share one config of your xyz file so we can check if things look correct.
